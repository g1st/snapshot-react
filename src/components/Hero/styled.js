import styled from 'styled-components';

export const Wrapper = styled.div`
  /* margin-top: 100px; */
  height: 100vh;
`;
Wrapper.displayName = 'Wrapper';
