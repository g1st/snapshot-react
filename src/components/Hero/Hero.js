import React from 'react';

import { Wrapper, LandingInfo, LandingText, HeaderText } from './styled';
import Button from '../Button';
import Background from './Background';

const Hero = () => {
  return (
    <Wrapper id="home">
      <Background />
      <LandingInfo>
        <HeaderText>Let's take a snapshot</HeaderText>
        <LandingText>
          Lorem, ipsum dolor sit amet consectetur adipisicing elit. Harum dicta
          repellendus veritatis commodi, voluptatem a facilis delectus.
        </LandingText>
        <Button title="Snapshot" big={false} />
      </LandingInfo>
    </Wrapper>
  );
};

export default Hero;
