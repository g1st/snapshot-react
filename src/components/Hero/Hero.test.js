import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Hero from '.';

describe('Hero Component', () => {
  it('matches snapshot', () => {
    const tree = shallow(<Hero />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it('Wrapper must have #home id attribute', () => {
    const tree = shallow(<Hero />);
    expect(tree.find('Wrapper').find('#home')).toHaveLength(1);
  });
});
