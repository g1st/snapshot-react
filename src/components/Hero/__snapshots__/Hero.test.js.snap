// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component matches snapshot 1`] = `
<Wrapper
  id="home"
>
  <Background />
  <LandingInfo>
    <HeaderText>
      Let's take a snapshot
    </HeaderText>
    <LandingText>
      Lorem, ipsum dolor sit amet consectetur adipisicing elit. Harum dicta repellendus veritatis commodi, voluptatem a facilis delectus.
    </LandingText>
    <Button
      big={false}
      title="Snapshot"
    />
  </LandingInfo>
</Wrapper>
`;
